src/envoy/utils/grpc_transport.cc

```diff
@@ -38,12 +38,20 @@ GrpcTransport<RequestType, ResponseType>::GrpcTransport(
     : async_client_(std::move(async_client)),
       response_(response),
       serialized_forward_attributes_(serialized_forward_attributes),
-      on_done_(on_done),
-      request_(async_client_->send(
-          descriptor(), request, *this, parent_span,
-          absl::optional<std::chrono::milliseconds>(kGrpcRequestTimeoutMs))) {
+      on_done_(on_done) {
   ENVOY_LOG(debug, "Sending {} request: {}", descriptor().name(),
             request.DebugString());
+  Envoy::Http::AsyncClient::RequestOptions options;
+  options.setTimeout(kGrpcRequestTimeoutMs);
+  Protobuf::RepeatedPtrField<envoy::api::v2::route::RouteAction::HashPolicy>
+      hash_policy;
+  hash_policy.Add()->mutable_header()->set_header_name(
+      kIstioAttributeHeader.get());
+  hash_policy.Add()->mutable_header()->set_header_name(
+      Envoy::Http::Headers::get().Host.get());
+  options.setHashPolicy(hash_policy);
```

Comment:
options.setHashPolicy(std::move(hash_policy))? Although I am not fully convinced if optimization indeed happens

Reply:
It's stack allocated and then copied into options. I'd think escape analysis decides to delete it right after this line.

Reply:
I agree with the former part and the escape detection. But I don't think compiler would do automatic move even not one is using it.
I will be astonished when compiler make this decision when it's a lock_guard-ish, so as to this.

Reply:
The argument is taken by reference. The only difference with a temporary is that deletion happens immediately rather than later, right? Either way, not clear if there's a measurable impact on performance.

Reply:
Aha, setHashPolicy is not a protobuf generated method. It doesn't have optimization for HashPolicy&& arg